### PR TITLE
Stocks

### PR DESCRIPTION
**Pull request by Bifabig**
- **Use Functional Components**
- **Implement lifecycle hooks**
- **Use store, actions and reducers in React**
- **Implement Routing using React Router**
- **Style using CSS modules**
- **Write unit tests with React Testing Library and Jest**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,220 @@
...
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app](https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app)
+## 🎥 Video <a name="video"></a>
 
-### Advanced Configuration
+<!-- > Add a link to your video project. -->
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+<a href="https://www.loom.com/share/15131936398e4c0397663968e4bc0e8c?sid=3244ce95-7795-46cf-aea9-afdbd3e0828c">Get an understanding of the project by watching this video</a>
+
```
Comment (Shedrack-Sunday) [inline README.md:90]: - [ ] In your video presentation, the last requirement was for your to highlight a piece of code you're proud of or a feature you're most proud of. I noticed you did talk about the search bar and also how you optimized the API with 1000s of values, but didn't show how it was implemented in your code. Kindly remake the video and highlight that section 👍 

### Requirements:
image

File: README.md [hunk 2]
```diff
@@ -1,70 +1,220 @@
...
+If you like this project:thumbsup:
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+<!-- > Give credit to everyone who inspired your codebase. -->
+
+Thanks to Microverse for this amazing project and Nelson Sakwa on Behance.
+
```
Comment (Shedrack-Sunday) [inline README.md:211]: - [ ] Nice job 👏 , However, kindly add the [appropriate link](https://www.behance.net/sakwadesignstudio) associated with the author 👍

File: src/redux/balanceSheets/balanceSheetsSlice.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export const getBalanceSheetItems = createAsyncThunk(
+  'balanceSheets/getBalanceSheetItems',
+  async (symbol, thunkAPI) => {
+    try {
+      const resp = await axios(`${balanceSheetsURL}${symbol.symbol}?apikey=b0f32ea981cc58fafa1177563647a780&limit=120`);
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(error);
```
Comment (Shedrack-Sunday) [inline src/redux/balanceSheets/balanceSheetsSlice.js:19]: - [ ] Based on project requirements, you meant to use a Filter stateless component to filter data retrieved from the API,  However, you didn't include the Filter stateless component. The code is focused on fetching `balancesheet` data using Redux and rendering it. There is no filtering functionality implemented in this component.

- [ ] To add filtering functionality, you need to create a separate component responsible for filtering the `balanceSheets` data. This new component should receive the `balanceSheets` data as props, apply the filtering logic, and render the filtered data.

### Sample code as a guide to you:
```
import React, { useState } from 'react';
import { useSelector } from 'react-redux';

const BalanceSheetFilter = () => {
  const balanceSheets = useSelector((state) => state.balanceSheets.balanceSheets);
  const [filteredBalanceSheets, setFilteredBalanceSheets] = useState([]);

  const handleFilter = (filterValue) => {
    const filteredData = balanceSheets.filter((balanceSheet) =>
      balanceSheet.someProperty === filterValue
    );
    setFilteredBalanceSheets(filteredData);
  };

  return (
    <div>
      {/* Render filter options and input */}
      {/* Add filter logic and update filteredBalanceSheets state */}
      {/* Render the filtered data */}
    </div>
  );
};

export default BalanceSheetFilter;
```

### Requirement
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Bifabig         👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Nice work with the setup✔️  
- Good use of redux for state management ✔️   
  
# Required Changes  🌵  

- [ ] You missed just a few of the requirements which I highlighted. Fix this and compliment your good job so far 👏 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Bifabig !

I've just realized that i reviewed your project without you meeting the GitHub flow requirement of pushing to the `main` branch from the `dev` branch. This is an error from me, and i sincerely apologize. 
I am sorry to mislead you. I haven't done that on purpose 👍 
- [ ] Kindly fix this issue if possible before the next review as the next reviewer will take note

- I apologise once more

**Comment by IngDavidTM** [comment]
This is the new PR: https://github.com/Bifabig/react-redux-capstone/pull/2#pullrequestreview-1508589374

**Pull request merged**